### PR TITLE
fix: Selected image get rotate in edit profile (#1270)

### LINKED ISSUE #1270: Selected image get rotate in edit profile
**Describe the bug**
Selected image get rotate in edit profile 

**To Reproduce**
1. Click pic from camera
2. Go to edit profile
3. select clicked pic from camera album

**Screenshots**
Screenshot_20190304-220421_Eventyay Attendee

**Smartphone Info:**

|Device         |Samsung Galaxy S7|
|Android Version  |Android 8.0.0   |

**Would you like to work on the issue?**
- [ ✔️] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by aman210697**
Fixes #1270 

Changes: 
Created RotateBitmap.kt file to prevent image from rotation
In EditProfileViewModel, created avatarUpdated, encodedImage, updatedImageTemp variable to retain the image.

Screenshots for the change:
Screenshot_20190311-160521_Eventyay Attendee

### CODE REVIEW

**Comment by aman210697** [comment]
> @liveHarshit @iamareebjamal @nikit19 could you review it?

**Comment by aman210697** [comment]
> > @liveHarshit @iamareebjamal @nikit19 could you review it?

**Comment by aman210697** [comment]
@iamareebjamal @liveHarshit @nikit19 please review this.

File: app/src/main/java/org/fossasia/openevent/general/RotateBitmap.kt [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import androidx.exifinterface.media.ExifInterface
+
+/**
+ * Created by aman210697 on 11/03/2019.
+ */
+
+class RotateBitmap {
+
+    private var mContext: Context? = null
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/RotateBitmap.kt]: We don't use hungarian notation

File: app/src/main/java/org/fossasia/openevent/general/RotateBitmap.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    /**
+     * This method is responsible for solving the rotation issue if exist. Also scale the images to
+     * 1024x1024 resolution
+     *
+     * @param selectedImage The Image URI
+     * @return Bitmap image results
+     * @throws IOException
+     */
+    @Throws(IOException::class)
+    fun HandleSamplingAndRotationBitmap(context: Context, selectedImage: Uri): Bitmap? {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/RotateBitmap.kt]: In which kotlin guideline does a function start with a capital character?

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 3]
```diff
@@ -18,22 +17,22 @@ import androidx.core.view.isVisible
...
 import com.google.android.material.snackbar.Snackbar
+import com.squareup.picasso.MemoryPolicy
 import com.squareup.picasso.Picasso
-import kotlinx.android.synthetic.main.fragment_edit_profile.view.editProfileCoordinatorLayout
-import kotlinx.android.synthetic.main.fragment_edit_profile.view.buttonUpdate
-import kotlinx.android.synthetic.main.fragment_edit_profile.view.firstName
-import kotlinx.android.synthetic.main.fragment_edit_profile.view.lastName
-import kotlinx.android.synthetic.main.fragment_edit_profile.view.profilePhoto
-import kotlinx.android.synthetic.main.fragment_edit_profile.view.progressBar
+import kotlinx.android.synthetic.main.fragment_edit_profile.view.*
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: No wildcard imports

Comment (aman210697) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: sure, IDE did itself

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt [hunk 4]
```diff
@@ -25,14 +26,22 @@ class EditProfileViewModel(
     private val mutableMessage = SingleLiveEvent<String>()
     val message: LiveData<String> = mutableMessage
 
+    var avatarUpdated = MutableLiveData<Boolean>(false)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: Variable `avatarUpdated` is not used in this view model.

Comment (aman210697) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: I used it in fragment and it should be normal variable not livedata, I'll update it

**Comment by aman210697** [comment]
@liveHarshit updated

File: app/schemas/org.fossasia.openevent.general.OpenEventDatabase/4.json [hunk 5]
```diff
@@ -0,0 +1,1430 @@
+{
```
Comment (liveHarshit) [inline app/schemas/org.fossasia.openevent.general.OpenEventDatabase/4.json]: Do not commit this file.

Comment (aman210697) [reply app/schemas/org.fossasia.openevent.general.OpenEventDatabase/4.json]: removed

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 6]
```diff
@@ -102,12 +94,25 @@ class EditProfileFragment : Fragment() {
...
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer { file ->
+                // prevent picasso from storing tempAvatar cache,
+                // if user select another image picasso will display tempAvatar instead of its own cache
+                    Picasso.get()
+                        .load(file)
+                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_person_black))
+                        .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
+                        .transform(CircleTransform())
+                        .into(rootView.profilePhoto)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: Wrong indentation

Comment (aman210697) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: fixed

**Pull request merged**